src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1118,6 +1155,312 @@ Containers::Optional<ImageData2D> AssimpImporter::doImage2D(const UnsignedInt id
+            /** @todo emulate this correctly by inserting keyframes
+                with the default node transformation */
+            return Animation::Extrapolation::Constant;
+        case aiAnimBehaviour_CONSTANT:
+            return Animation::Extrapolation::Constant;
+        case aiAnimBehaviour_LINEAR:
+            return Animation::Extrapolation::Extrapolated;
+        case aiAnimBehaviour_REPEAT:
+            return Animation::Extrapolation::Constant;
+        default: CORRADE_INTERNAL_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */
+    }
+}
+
+}
+
+Containers::Optional<AnimationData> AssimpImporter::doAnimation(UnsignedInt id) {
+    /* Import either a single animation or all of them together. At the moment,
+       Blender doesn't really support cinematic animations (affecting multiple
+       objects): https://blender.stackexchange.com/q/5689. And since
+       https://github.com/KhronosGroup/glTF-Blender-Exporter/pull/166, these
+       are exported as a set of object-specific clips, which may not be wanted,
+       so we give the users an option to merge them all together. */
+    const std::size_t animationBegin =
+        configuration().value<bool>("mergeAnimationClips") ? 0 : id;
+    const std::size_t animationEnd =
+        configuration().value<bool>("mergeAnimationClips") ? _f->scene->mNumAnimations : id + 1;
+
+    /* Calculate total channel count */
+    std::size_t channelCount = 0;
+    for(std::size_t a = animationBegin; a != animationEnd; ++a) {
+        channelCount += _f->scene->mAnimations[a]->mNumChannels;
+    }
+
+    /* Presence of translation/rotation/scaling keys for each channel.
+       We might skip certain tracks (see comments in the next loop)
+       but we need to know the correct track count and data array size
+       up front, so determine this and remember it for the actual
+       loop that extracts the tracks. */
+    Containers::Array<std::tuple<bool, bool, bool>> channelTrackPresence{channelCount};
+
+    /* Calculate total data size and track count. If merging all animations together,
+       this is the sum of all clip track counts. */
+    std::size_t dataSize = 0;
+    std::size_t trackCount = 0;
+    std::size_t currentChannel = 0;
+    for(std::size_t a = animationBegin; a != animationEnd; ++a) {
+        const aiAnimation* animation = _f->scene->mAnimations[a];
+        for(std::size_t c = 0; c != animation->mNumChannels; ++c) {
+            const aiNodeAnim* channel = animation->mChannels[c];
+
+            UnsignedInt translationKeyCount = channel->mNumPositionKeys;
+            UnsignedInt rotationKeyCount = channel->mNumRotationKeys;
+            UnsignedInt scalingKeyCount = channel->mNumScalingKeys;
+
+            /* Assimp adds useless 1-key tracks with default node
+               translation/rotation/scale if the channel doesn't
+               target all 3 of them. Ignore those. */
+            if(translationKeyCount == 1 || rotationKeyCount == 1 || scalingKeyCount == 1) {
```

Comment:
Should this be hidden behind a config value? Maybe something generic like `removeEmptyAnimations` that would also remove completely empty tracks (that would break the glTF test, but I don't see a good real-world use-case for that anyway).

Reply:
Config value (that's enabled by default) sounds like a good idea :+1: Turning the option off might also give the users a fallback option in case this postprocessing turns out to have some unhandled corner case and provides wrong output (or if they need to be bug-compatible with Assimp for whatever other reason).

About empty tracks, I can see a use case where some postprocessing/optimization pipeline removes all keyframes because they're redundant / identity but the end-user app still expects to look up that particular animation under a name and play it. So I would personally keep them.

Reply:
I added `removeDummyAnimationTracks` (+ test and docs) and kept empty tracks completely untouched. A catch-all config value for both seemed silly, and so did an extra one just for cleanup.